### ISSUE TITLE
[🐛 Bug]: ensure python uses SE_MANAGER_PATH as a pathlib.Path

### ISSUE THREAD

**Issue by bollwyvl** [SeleniumHQ/selenium#12929]
### What happened?

When `SE_MANAGER_PATH` is set, the python implementation uses it directly as a `str` rather than a `Path`, but then tries to use `Path.is_file`.

### How can we reproduce the issue?

```shell
import os
os.environ["SE_MANAGER_PATH"] = "some/actual/path/to/an/existing/selenium-manager"
from selenium.webdriver.common.selenium_manager import SeleniumManager
assert SeleniumManager.get_binary().exists()
```

### Relevant log output

```shell
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File ".../lib/python3.12/site-packages/selenium/webdriver/common/selenium_manager.py", line 60, in get_binary
    if not path.is_file() and os.environ["CONDA_PREFIX"]:
           ^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'is_file'
```

### Operating System

CentOS7

### Selenium version

4.14.0

### What are the browser(s) and version(s) where you see this issue?

-

### What are the browser driver(s) and version(s) where you see this issue?

-

### Are you using Selenium Grid?

-

**Comment by symonk** [comment]
Thanks!

**Comment by bollwyvl** [comment]
I _think_ the patch in [e15fed5](https://github.com/aguspe/selenium/commit/e15fed519d1035a8a0701ce483dd225e2694fdb6) was missing some parens: sorry I didn't catch it initially, I don't have a lot of walrus experience. We shipped this on https://github.com/conda-forge/selenium-feedstock/pull/69

```diff
--- a/selenium/webdriver/common/selenium_manager.py
+++ b/selenium/webdriver/common/selenium_manager.py
@@ -41,7 +41,7 @@ class SeleniumManager:
         :Returns: The Selenium Manager executable location
         """
 
-        if path := os.getenv("SE_MANAGER_PATH") is not None:
+        if (path := os.getenv("SE_MANAGER_PATH")) is not None:
             return Path(path)
         else:
             platform = sys.platform
```

**Comment by titusfortner** [comment]
@symonk / @isaulv / @AutomatedTester can someone verify the logic above? I can get a patch release out if what we have isn't correct.

**Comment by isaulv** [comment]
@titusfortner verified that you need parentheses. Without it, `path` becomes `True`.

**Comment by titusfortner** [comment]
fixed, releasing 4.15.1
